staging/src/k8s.io/client-go/util/watchlist/watch_list.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watchlist
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientfeatures "k8s.io/client-go/features"
+)
+
+// CanUseWatchListForListRequest determines if the given
+// ListOptions can be used to form a watch-list request,
+// which would result in streaming data from the server.
+func CanUseWatchListForListRequest(opts metav1.ListOptions) bool {
+	if !clientfeatures.FeatureGates().Enabled(clientfeatures.WatchListClient) {
+		return false
+	}
+
+	if len(opts.Continue) > 0 {
```

Comment:
I would like to avoid logic duplication here - it's easy to get out-of-sync here.

Instead of that, can we just do sth like:
```
opts.SendInitialEvents = true
opts.Watch = true
if ValidateListOptions(opts) != nil {
  return false
}
return true
```

In general, if some combination is not valid for watch-list, our validation should reject that and we should reuse the fact of it.

Reply:
created a pr that shows what would be needed https://github.com/kubernetes/kubernetes/pull/125440 for ^